src/validation_workflow/docker/validation_docker.py

```diff
@@ -151,12 +163,29 @@ def run_container(self, OpenSearch_image: str, OpenSearchDashboard_image: str, v
             self.source_file = os.path.join('docker', 'release', 'dockercomposefiles', self.docker_compose_files[self.version_number])
             shutil.copy2(self.source_file, self.target_yml_file)
 
-        self.inplace_change(self.target_yml_file, f'opensearchproject/opensearch:{version[0]}', f'{OpenSearch_image}:{version}')
-        self.inplace_change(self.target_yml_file, f'opensearchproject/opensearch-dashboards:{version[0]}', f'{OpenSearchDashboard_image}:{version}')
-
+        if (self.args.using_staging_artifact_only):
+            self.images_used = f'{OpenSearch_image}:{version}'
+        else:
+            self.images_used = f'{OpenSearch_image}:{version}'
+
+        self.inplace_change(
+            self.target_yml_file,
+            f'opensearchproject/opensearch:{version[0]}',
+            f'{OpenSearch_image}:{version}.{self.args.os_build_number}'
+            if self.args.using_staging_artifact_only
+            else f'{OpenSearch_image}:{version}'
+        )
+
+        self.inplace_change(
+            self.target_yml_file,
+            f'opensearchproject/opensearch-dashboards:{version[0]}',
+            f'{OpenSearchDashboard_image}:{version}.{self.args.osd_build_number}'
+            if self.args.using_staging_artifact_only
+            else f'{OpenSearchDashboard_image}:{version}'
+        )
```

Comment:
The inplace change is actually saving on the original file?
Or saved on a temp file that automatically get cleaned up later?

Reply:
Thanks for the question, @peterzhuamazon .. it does the inplace_change on the copied file stored at temp folder. The source file is not toched.